File: Tools/QS2.license.core/EMail.vb


Public Class EMail


    Public Shared Sub openNewEMail(ByVal sTo As String, Optional ByVal sTitle As String = "", Optional ByVal sText As String = "")

        Dim sParam As String

        ' Betreff
        If sTitle.Length > 0 Then AddMailParam(sParam, "subject=" + sTitle)

        ' Nachrichtentext
        If sText.Length > 0 Then AddMailParam(sParam, "body=" + sText.Replace(Chr(10), "%0d").Replace(Chr(13), "%0a"))

        ' Fenster "Neue Nachricht" öffnen
        Process.Start("mailto: " + sTo + sParam)

    End Sub
    Private Shared Sub AddMailParam(ByRef sAllParam As String, ByVal sParam As String)
        If sAllParam = String.Empty Then
            sAllParam = "?" + sParam
        Else
            sAllParam &= "&" + sParam
        End If
    End Sub



End Class

File: Tools/QS2.functions.cs.obsolet/EMail.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Windows.Forms;



namespace QS2.functions.cs
{

    public class EMail
    {
        
        public static string EMailService { get; set; }= "ServiceCenter@s2-engineering.com";



        public void sendEMail(string sTitle, string sText, string sTo)
        {
            string sParam = "";
            if (sTitle.Length > 0)
            {
                this.addMailParam(ref sParam, "subject=" + sTitle);
            }
            if (sTitle.Length > 0)
            {
                string sTxtNew = sText.Replace(Convert.ToChar(10).ToString(), "%0d");
                sTxtNew = sText.Replace(Convert.ToChar(13).ToString(), "%0a");
                this.addMailParam(ref sParam, "body=" + sTxtNew);
            }

            System.Diagnostics.Process.Start("mailto: " + sTo + sParam);
        }
        public void addMailParam(ref string sAllParam, string sParam)
        {
            if (sAllParam == string.Empty)
            {
                sAllParam += "?" + sParam;
            }
            else
            {
                sAllParam += "&" + sParam;
            }
        }


    }


}
